akka-http-tests/src/test/scala/akka/http/scaladsl/server/DiscardEntityDefaultRejectionHandlerSpec.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.scaladsl.server
+
+import akka.http.scaladsl.model.ContentTypes.`text/plain(UTF-8)`
+import akka.http.scaladsl.model.HttpEntity
+import akka.http.scaladsl.model.StatusCodes.NotFound
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+import org.scalatest.concurrent.Eventually._
+import org.scalatest.concurrent.ScalaFutures
+
+class DiscardEntityDefaultRejectionHandlerSpec extends RoutingSpec with ScalaFutures {
+
+  private val route = path("foo") {
+    complete("bar")
+  }
+
+  private var elementsEmitted = 0
+  private def gimmeElement(): ByteString = {
+    elementsEmitted = elementsEmitted + 1
+    ByteString("Foo")
+  }
+
+  private val ThousandElements: Stream[ByteString] = Stream.continually(gimmeElement()).take(1000)
+  private val RequestToNotHandled = Get("/bar", HttpEntity(`text/plain(UTF-8)`, Source[ByteString](ThousandElements)))
+
+  "Default RejectionHandler" should {
+    "rejectEntity by default" in {
+      RequestToNotHandled ~> Route.seal(route) ~> check {
+        status shouldBe NotFound
+        eventually {
+          elementsEmitted should be > 1
```

Comment:
Hmm. So when the rejection handler is triggered, it will still have the client send the entire body? Why not check exactly that rather than >1 then?

Reply:
As far as I understand, all the bytes will be discarded, so yes, all the bytes should be sent from the client.
You are right, I can check for the exact number.

Reply:
I know early response is problematic, but this feels a bit weird, let's say I was going to do an upload with a gigabyte of data but used the wrong path...

Reply:
mmm maybe I misunderstood the problem stated in https://github.com/akka/akka-http/issues/38.
If the entity is not discarded, the connections will stall, right?

Reply:
Please let me know if I got it wrong

Reply:
Yeah, not sure what is right here, just struck me that it sounds like weird behavior. Gut feeling is that we'd rather want to cancel, but I guess then the entire connection is torn down?